wait/host_port.go

```diff
@@ -95,6 +95,25 @@ func (hp *HostPortStrategy) Timeout() *time.Duration {
 	return hp.timeout
 }
 
+// detectInternalPort returns the lowest internal port that is currently bound.
+// If no internal port is found, it returns the zero nat.Port value which
+// can be checked against an empty string.
+func (hp *HostPortStrategy) detectInternalPort(ctx context.Context, target StrategyTarget) (nat.Port, error) {
+	var internalPort nat.Port
+	inspect, err := target.Inspect(ctx)
+	if err != nil {
+		return internalPort, fmt.Errorf("inspect: %w", err)
+	}
+
+	for port := range inspect.NetworkSettings.Ports {
+		if internalPort == "" || port.Int() < internalPort.Int() {
+			internalPort = port
```

Comment:
suggestion: can we just break after setting the internal port to avoid entering the loop again?

Reply:
We're looking for the lowest port which might not be first port, so not if we want to maintain the that current behaviour. This was introduced in this fix https://github.com/testcontainers/testcontainers-go/pull/2641